tac/agents/controller/agent.py

```diff
@@ -200,7 +200,6 @@ def stop(self) -> None:
         if self.game_handler.game_phase == GamePhase.GAME or self.game_handler.game_phase == GamePhase.GAME_SETUP:
             self.game_handler.notify_competition_cancelled()
         super().stop()
-        self.teardown()
```

Comment:
So where is teardown called? In super?

Reply:
yes. `super().stop()` calls `self.teardown()`